src/python/pants/core/subsystems/python_bootstrap.py

```diff
@@ -288,4 +286,5 @@ def rules():
     return (
         *collect_rules(),
         *asdf.rules(),
+        *add_option_fields_for(PythonBootstrapSubsystem),
```

Comment:
It might be better if this is something like `PythonBootstrapSubsystem.environment_target_rules()`. 

Or even cooler would be that we automatically register it when creating the `SubsystemRule`. But that would be better as a follow up.

Reply:
> Or even cooler would be that we automatically register it when creating the SubsystemRule. But that would be better as a follow up.

Yea, this is probably the way to go. I don't think that it would be that complex though, so it might be good to do it here rather than in a followup?

Would basically involve changing this method to return a list: https://github.com/pantsbuild/pants/blob/7d7eb221ec9c2eba6395c00cde316cddb344eeff/src/python/pants/engine/rules.py#L45-L54 ... or perhaps deprecating it, and mixing @Eric-Arellano's suggestions and replacing `SubsystemRule(PythonBootstrapSubsystem)` with `PythonBootstrapSubsystem.rules()`.

Reply:
Ohhh, `PythonBootstrapSubsystem.rules()` could be good. Note that it has been highly confusing to plugin authors that subsystems are only registered via `collect_rules()` if that subsystem is consumed by an `@rule`. So when first debugging, you have to do `SubsystemRule(PythonBootstrapSubsystem)`.

I like that `PythonBootstrapSubsystem.rules()` avoids you needing to import anything, so it makes it a little more feasible to us start saying you should always add `MySubsystem.rules()` to `register.py`

Reply:
This general change sounds good -- the surface area is going to be much bigger than this specific PR, so I think I'd rather take it as a fast-follow, if that's OK?

Reply:
I don't think that the surface area is any larger if you don't actually execute the deprecation of the previous way of doing things in this PR. But up to you.